tests/testingResources.go

```diff
@@ -0,0 +1,329 @@
+	if deploy.Status.AvailableReplicas < 1 {
+		var i int = 0
+		for deploy.Status.AvailableReplicas < 1 {
+			fmt.Printf("======= ExternalDNS not available, checking again in %s seconds ====", timeout)
+			time.Sleep(timeout)
+			i++
+
+			if i >= 5 {
+				extDNSReady = false
+			}
+		}
+	}
+
+	if extDNSReady {
+		lgr.Info("External Dns deployment is running and ready")
+		return nil
+	} else {
+		return fmt.Errorf("external dns deployment is not running in pod, check logs")
+	}
+
+}
+
+func RunCommand(ctx context.Context, subId, rg, clusterName string, request armcontainerservice.RunCommandRequest, opt runCommandOpts) (armcontainerservice.CommandResultProperties, error) {
+	fmt.Println("IN RUN COMMAND for command for Tests +++++++++++++ command: ", *request.Command)
+
+	lgr := logger.FromContext(ctx)
+	ctx = logger.WithContext(ctx, lgr)
+
+	lgr.Info("starting to run command")
+	defer lgr.Info("finished running command for testing")
+
+	emptyResp := &armcontainerservice.CommandResultProperties{}
+	//fmt.Println("#1 Before getting az creds")
+	cred, err := getAzCred()
+	if err != nil {
+		return *emptyResp, fmt.Errorf("getting az credentials: %w", err)
+	}
+
+	//fmt.Println("#2 Before creating managed cluster client")
+	client, err := armcontainerservice.NewManagedClustersClient(subId, cred, nil)
+	if err != nil {
+		return *emptyResp, fmt.Errorf("creating aks client: %w", err)
+	}
+
+	//fmt.Println("#3 Before BeginRunCommand() call")
+	poller, err := client.BeginRunCommand(ctx, rg, clusterName, request, nil)
+	if err != nil {
+		return *emptyResp, fmt.Errorf("starting run command: %w", err)
+	}
+
+	//fmt.Println("#4 Before Poller PollUnitlDone() call")
+	result, err := poller.PollUntilDone(ctx, nil)
+	if err != nil {
+		return *emptyResp, fmt.Errorf("running command: %w", err)
+	}
+
+	//fmt.Println("#5 Before Checking Logs")
+	logs := ""
+	if result.Properties != nil && result.Properties.Logs != nil {
+		logs = *result.Properties.Logs
+		//fmt.Println("logs from run command in testing: ", logs)
+	}
+
+	//fmt.Println("#5 Before output file code")
+	if opt.outputFile != "" {
+
+		outputFile, err := os.OpenFile(opt.outputFile, os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0644)
+		if err != nil {
+
+			return *result.Properties, fmt.Errorf("creating output file %s: %w", opt.outputFile, err)
+		}
+		defer outputFile.Close()
+		_, err = outputFile.WriteString(logs)
+		if err != nil {
+			return *result.Properties, fmt.Errorf("writing output file %s: %w", opt.outputFile, err)
+		}
+	} else {
+		lgr.Info("using command logs, no output file specified")
+	}
+
+	//fmt.Println("#6 Before checking exit code")
+	if *result.Properties.ExitCode != 0 {
+		lgr.Info(fmt.Sprintf("command failed with exit code %d", *result.Properties.ExitCode))
+		return *result.Properties, nonZeroExitCode
+	}
+
+	//fmt.Println("returning logs with no error >>>>>>>>>>>>>>>>>>>>>")
+	return *result.Properties, nil
+}
+
+func getAzCred() (azcore.TokenCredential, error) {
```

Comment:
You can probably remove this function since you use the one from the `clients` package instead